app/services/files_service.ts

```diff
@@ -10,13 +10,41 @@ import {
   FileServiceFileReadError,
   FileServiceFileUploadError,
 } from "#exceptions/file_service_errors";
+import FileEntry from "#services/file_entry";
+
+interface FileInformation {
+  fileInstance: FileEntry;
+  diskKey: string;
+}
 
 export default class FilesService {
-  private generateKey(extname: string | undefined): string {
-    if (extname?.length === 0) {
-      extname = undefined;
-    }
-    return `${randomUUID()}.${extname ?? "bin"}`;
+  private static getExtension(extname: string | undefined): string {
+    return extname !== undefined && extname.length > 0 ? extname : "bin";
+  }
+
+  private static mapToFileInstance(extname: string | undefined): FileEntry {
+    const fileEntry = new FileEntry();
+    fileEntry.id = randomUUID();
+    fileEntry.fileType = this.getExtension(extname);
+    return fileEntry;
+  }
```

Comment:
`mapToFileInstance` seems like the wrong name for this function, as it allocates a new file entry instead of mapping an existing one to a model instance.
also maybe this should be a factory method on `FileEntry` instead?